### PR TITLE
Create release pipeline

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Added a pipeline that allows to re-tag `server`, `admin-tools` and `auto-setup` images and push to temporalio

## Why?

Facilitates release process

## Checklist

1. Closes 

2. How was this tested:

executed with `act` `echo`ing the last skope line: 
```
| skopeo copy --all docker://temporaliotest/ui:b56843785892b64deae8f08d848ca8ae997c24dc docker://temporaliotest/ui:tag
| skopeo copy --all docker://temporaliotest/ui-test:b56843785892b64deae8f08d848ca8ae997c24dc docker://temporaliotest/ui-test
```
3. Any docs updates needed?

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Set up skopeo
+        run: |
+          skopeo() {
+            # Needed for older docker versions
```
Comment (dnr) [inline .github/workflows/release.yml:25]: have you tried without this? the docker version on gha might be new enough

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            # See: https://github.com/containers/skopeo/issues/1501 and https://github.com/moby/moby/issues/42680
+            local seccomp="--security-opt seccomp=unconfined"
+
+            docker run --rm -i $seccomp quay.io/skopeo/stable "$@"
+          }
+          skopeo login -u ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_PASSWORD }} docker.io
```
Comment (dnr) [inline .github/workflows/release.yml:31]: ideally, use env vars for these so you can do the quoting properly.

(but at the very least, put the substitutions in single quotes)

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+      - name: Copy images
+        env:
+          IMAGES: server auto-setup admin-tools
+          SRC_REPO: temporaliotest
+          DST_REPO: temporalio
+          SRC_TAG: ${{ steps.calc_args.outputs.image_tag }}
+          DST_TAG: ${{ github.event.inputs.tag }}
+        run: |
+          for image in ${IMAGES//,/ }; do
```
Comment (dnr) [inline .github/workflows/release.yml:48]: you don't need the comma to space substitution if you're setting the env var right here

**Review by dnr — COMMENTED** [review]
I thought GHA had a built-in or vetted community action to do this, so we wouldn't need to write our own scripts?

Can you point to a sample run output? I'm wondering if the shell function thing works across steps. (Obviously not to the main repo, but maybe just from temporaliotest to temporaliotest)

File: .github/workflows/release.yml (hunk 1 above)
Comment (feedmeapples) [reply .github/workflows/release.yml:25]: actually wanted to ask you how this was used in the other pipeline

**Comment by feedmeapples** [comment]
working with the security team to define whether we can use a GHA like https://github.com/giantswarm/retagger

**Comment by feedmeapples** [comment]
> working with the security team to define whether we can use a GHA like https://github.com/giantswarm/retagger

added https://github.com/temporalio/docker-builds/pull/29

**Comment by feedmeapples** [comment]
> I thought GHA had a built-in or vetted community action to do this, so we wouldn't need to write our own scripts?
> 
> Can you point to a sample run output? I'm wondering if the shell function thing works across steps. (Obviously not to the main repo, but maybe just from temporaliotest to temporaliotest)

i've checked several, actions that specifically re-tag images at most have dozens of stars and mostly do the same as the `skopeo` script, maybe using `crane` from google instead
these are the ones i've found
https://github.com/koraykoska/retag-docker-image
https://github.com/akhilerm/tag-push-action
https://github.com/abronin/ecr-retag-action
https://github.com/tinact/docker.image-retag

Doesn't seem to be worth it from security or UX in case of forking, since the manual `skopeo` is still pretty short and easy to read

**Review by dnr — APPROVED** [review]
You could run a test just copying to a different tag in `temporaliotest`

**Comment by feedmeapples** [comment]
tested on https://hub.docker.com/layers/temporaliotest/server/sha-315d969/images/sha256-31c60c8fd70f66b533551c5e4f90e13a36187f53bcae7b8fabe65def26b23707?context=explore

resulting image:
https://hub.docker.com/layers/temporaliotest/server/test/images/sha256-31c60c8fd70f66b533551c5e4f90e13a36187f53bcae7b8fabe65def26b23707?context=explore

hashes match

**Pull request merged**